tests/test_data_objects.py

```diff
@@ -181,3 +182,27 @@ def test_spans_intersection(
 
     intersection = span1.intersect(span2, ignore_entity_type=ignore_entity_type)
     assert intersection == intersection_length
+
+
+@pytest.mark.parametrize(
+    "start1, end1, start2, end2, expected_overlap_ratio, ignore_entity_type",
+    [
+        (150, 153, 160, 165, 0.0, True),
+        (150, 153, 150, 153, 1.0, True),
+        (150, 153, 152, 154, 0.4, True),
+        (150, 153, 100, 151, 0.04, True),
+        (150, 153, 100, 151, 0.0, False),
```

Comment:
Although the ratio calculation between [150, 153] and [100, 151] is not 0, but in this case the returned overlap is 0 because the ignore_entity_type=False (ignore_entity_type value is in the last value of @pytest.mark.parametrize)
So the test function has already covered this case **Could we add a test case for overlap + ignore_entity_type=False?**